### PR TITLE
Mixpanel: Filtering out individual items based on datetime in state

### PR DESCRIPTION
**Pull request by Kopiczek**
## What
Mixpanel export API allow to filter based on DATE (without the time portion).
This means that if a sync is run multiple times a day, we keep duplicating the same data over and over.
This change filters the records after getting them from source.

## How
This is done on export stream only by comparing the cursor field of each record to stream state.

## Recommended reading order
1. `exports.py`
2. `base.py`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
</details>

## Tests

<details><summary><strong>Unit</strong></summary>

Test that verifies that a record is filtered by state.

</details>

<details><summary><strong>Integration</strong></summary>

N/A

</details>

<details><summary><strong>Acceptance</strong></summary>

N/A

</details>

### CODE REVIEW

**Comment by Kopiczek** [comment]
@marcosmarxm am I doing something wrong trying to initialize those test you did?

**Comment by marcosmarxm** [comment]
@Kopiczek sorry the delay here, I'll take a look.

File: airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py [hunk 1]
```diff
@@ -426,10 +426,15 @@ def test_export_stream(requests_mock, export_response):
...
     # read records for single slice
     records = stream.read_records(sync_mode=SyncMode.incremental, stream_slice=stream_slice)
 
     records_length = sum(1 for _ in records)
     assert records_length == 1
+
+    # no records should be returned when state older than the record
+    records = stream.read_records(sync_mode=SyncMode.incremental, stream_slice=stream_slice, stream_state=stream_state)
+    records_length = sum(1 for _ in records)
+    assert records_length == 0
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py:440]: I don't think this is going to work with unit test and also this is not following the best practices. Please create another test to validate what you're implementing.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Unit test are broken @Kopiczek please implement a new unit test

File: airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py (hunk 1 above)
Comment (Kopiczek) [reply airbyte-integrations/connectors/source-mixpanel/unit_tests/test_streams.py]: Yes, forgot to push changed tests after the implementation was modified (v1 was doing client side filtering)

**Comment by Kopiczek** [comment]
> Unit test are broken @Kopiczek please implement a new unit test

This should be working now. All unit tests passing locally

**Comment by Kopiczek** [comment]
Sorry for all the back and forth.
Since the code treats the long date from source as current timezone we are doing the same when going from cursor to long
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/export.py#L122-L124

We do the same, forgot to include that in the test and hardcoded my local timezone into expected result

**Comment by marcosmarxm** [comment]
@roman-yermilov-gl can you do the final review here?

**Review by roman-yermilov-gl — APPROVED** [review]

**Comment by Kopiczek** [comment]
@marcosmarxm need your approval as well since you requested changes.

**Comment by marcosmarxm** [comment]
> @marcosmarxm need your approval as well since you requested changes.

@Kopiczek can you bump the Dockerfile version and documentation, I'll publish after it

**Comment by Kopiczek** [comment]
@marcosmarxm done

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**